packages/sdk/src/filter/filter.ts

```diff
@@ -1,297 +1,259 @@
-  private connectionManager: ConnectionManager;
-  private libp2p: Libp2p;
-  private peerManager: PeerManager;
-  private lightPush?: ILightPush;
-  private activeSubscriptions = new Map<string, Subscription>();
+  private subscriptions = new Map<PubsubTopic, Subscription>();
 
   public constructor(params: FilterConstructorParams) {
-    this.config = buildConfig(params.options);
-    this.lightPush = params.lightPush;
-    this.peerManager = params.peerManager;
+    this.config = {
+      numPeersToUse: 2,
+      pingsBeforePeerRenewed: 3,
+      keepAliveIntervalMs: 60_000,
+      ...params.options
+    };
+
     this.libp2p = params.libp2p;
+    this.peerManager = params.peerManager;
     this.connectionManager = params.connectionManager;
 
     this.protocol = new FilterCore(
-      async (pubsubTopic, wakuMessage, peerIdStr) => {
-        const subscription = this.getActiveSubscription(pubsubTopic);
-        if (!subscription) {
-          log.error(
-            `No subscription locally registered for topic ${pubsubTopic}`
-          );
-          return;
-        }
-
-        await subscription.processIncomingMessage(wakuMessage, peerIdStr);
-      },
-
+      this.onIncomingMessage.bind(this),
       params.connectionManager.pubsubTopics,
       params.libp2p
     );
-
-    this.activeSubscriptions = new Map();
   }
 
   public get multicodec(): string {
     return this.protocol.multicodec;
   }
 
   /**
-   * Opens a subscription with the Filter protocol using the provided decoders and callback.
-   * This method combines the functionality of creating a subscription and subscribing to it.
+   * Unsubscribes from all active subscriptions across all pubsub topics.
    *
-   * @param {IDecoder<T> | IDecoder<T>[]} decoders - A single decoder or an array of decoders to use for decoding messages.
-   * @param {Callback<T>} callback - The callback function to be invoked with decoded messages.
+   * @example
+   * // Clean up all subscriptions when React component unmounts
+   * useEffect(() => {
+   *   return () => filter.unsubscribeAll();
+   * }, [filter]);
    *
-   * @returns {Promise<SubscribeResult>} A promise that resolves to an object containing:
-   *   - subscription: The created subscription object if successful, or null if failed.
-   *   - error: A ProtocolError if the subscription creation failed, or null if successful.
-   *   - results: An object containing arrays of failures and successes from the subscription process.
-   *     Only present if the subscription was created successfully.
+   * @example
+   * // Reset subscriptions and start over
+   * filter.unsubscribeAll();
+   * await filter.subscribe(newDecoder, newCallback);
+   */
+  public unsubscribeAll(): void {
+    for (const subscription of this.subscriptions.values()) {
+      subscription.stop();
+    }
+
+    this.subscriptions.clear();
+  }
+
+  /**
+   * Subscribes to messages with specified decoders and executes callback when a message is received.
```

Comment:
this seems to be a repeat of the js docs for the filter interface, can we just have it in either one or the other?

Reply:
that's right, removing js-doc from impl